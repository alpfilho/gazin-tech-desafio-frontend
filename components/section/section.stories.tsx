import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Section } from './section';

export default {
	title: 'Layout/Section',
	component: Section,
	parameters: {
		layout: 'fullscreen'
	}
} as ComponentMeta<typeof Section>;

export const Default: ComponentStory<typeof Section> = () => (
	<Section>
		<h1>Example Content</h1>
	</Section>
);
