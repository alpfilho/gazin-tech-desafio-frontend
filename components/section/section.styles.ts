import styled from 'styled-components';

import { Container, Content } from 'components/contentContainer';

export const SectionContainer = styled.section`
	padding-top: 2em;

	${Container} {
		${Content} {
			display: flex;
			align-items: flex-start;
			justify-content: center;
			flex-direction: column;
		}
	}
`;
