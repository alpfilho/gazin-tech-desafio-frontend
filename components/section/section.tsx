import React from 'react';

import { SectionContainer } from './section.styles';

export const Section: React.FC = ({ children }) => {
	return <SectionContainer>{children}</SectionContainer>;
};
