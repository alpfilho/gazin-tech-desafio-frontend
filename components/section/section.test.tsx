import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from './section.stories';

const { Default } = composeStories(stories);

describe('Section', () => {
	it('render a section', () => {
		render(<Default />);

		const exampleContent = screen.getByText('Example Content');
		expect(exampleContent).toBeInTheDocument();
	});

	it('section render unchanged', () => {
		const { container } = render(<Default />);

		expect(container).toMatchSnapshot();
	});
});
