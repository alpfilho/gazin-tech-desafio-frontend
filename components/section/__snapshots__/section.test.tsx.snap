// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section section render unchanged 1`] = `
<div>
  <section
    class="SectionContainer-sc-cxauh6-0 eLHtVq"
  >
    <h1>
      Example Content
    </h1>
  </section>
</div>
`;
